File: src/linkedlist/lc_21_merge_two_sorted_list/merge_two_sorted_list.java
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode t1, t2;
        t1 = list1;
        t2 = list2;
        ListNode dummy = new ListNode(-1);
        ListNode temp = dummy;
        while(t1 != null && t2 != null){
            if(t1.val <= t2.val) {
                temp.next = t1;
                t1 = t1.next;
            } else {
                temp.next = t2;
                t2 = t2.next;
            }
            temp = temp.next;
        }
        while(t1 != null){
            temp.next = t1;
            t1 = t1.next;
            temp = temp.next;
        }
        while(t2 != null){
            temp.next = t2;
            t2 = t2.next;
            temp = temp.next;
        }
        return dummy.next;
    }
}

File: src/linkedlist/lc_21_merge_two_sorted_list/merge_two_sorted_list.py
from listNode import ListNode
class Solution:
    def mergeTwoLists(self, list1: ListNode, list2: ListNode) -> ListNode:
        p1, p2 = list1, list2
        dummy = ListNode(0)
        temp = dummy
        while p1 and p2:
            if p1.val <= p2.val:
                temp.next = ListNode(p1.val)
                p1 = p1.next
            else:
                temp.next = ListNode(p2.val)
                p2 = p2.next
            temp = temp.next
        if p1:
            while p1:
                temp.next = ListNode(p1.val)
                p1 = p1.next
                temp = temp.next
        if p2:
            while p2:
                temp.next = ListNode(p2.val)
                p2 = p2.next
                temp = temp.next
        return dummy.next